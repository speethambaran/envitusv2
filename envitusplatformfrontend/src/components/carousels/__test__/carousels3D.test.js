import React from 'react';
import { shallow } from 'enzyme';
import Carousels3D from './../carousels3D';
import { findByDataTest } from './../../../Utils';

const preConfig = (props={}) => {
    const component = shallow(<Carousels3D {...props} />);
    return component;
};

describe('Carousels3D Test', () => {
    it('Redering Properly with prop values', () => {
        let component;
        const props = {
            items: [{
                id: "testBuilding",
                img: "dummypath",
                alt: "testAltBuilding"
            }] 
        };
        component = preConfig(props);
        const wrapperMV = findByDataTest(component, 'modelViewComponent');
        expect(wrapperMV.length).toBe(1);
        const wrapperDevice = findByDataTest(component, 'DeviceComponent');
        expect(wrapperDevice.length).toBe(1);
        const wrapperDiv = findByDataTest(component, 'mainDiv');
        expect(wrapperDiv.length).toBe(1);
        expect(wrapperDiv).toMatchSnapshot();
    });

});