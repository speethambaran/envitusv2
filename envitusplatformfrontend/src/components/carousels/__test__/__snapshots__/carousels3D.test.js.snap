// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousels3D Test Redering Properly with prop values 1`] = `
<div
  data-test="mainDiv"
>
  <model-viewer
    alt="testAltBuilding"
    auto-rotate={true}
    camera-controls={true}
    className="d-block mx-auto"
    data-test="modelViewComponent"
    key="testBuilding"
    src="dummypath"
  >
    <Connect(DeviceList)
      data-test="DeviceComponent"
      listType="sensor3D"
    />
  </model-viewer>
</div>
`;
